packages/react-icons/src/utils/createSvgIcon.ts

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import { css, getNativeProps, htmlElementProperties } from '@uifabric/utilities';
+import { classes, MS_ICON } from './SvgIcon.styles';
+import { ISvgIconProps } from './SvgIcon.types';
+import { SvgIconCreateFnParams } from './types';
+
+const createSvgIcon = <TProps = {}>({ svg, displayName }: SvgIconCreateFnParams<TProps>) => {
+  const Component: React.FC<React.HTMLAttributes<HTMLSpanElement> & TProps & ISvgIconProps> = props => {
+    const { className, style = {} } = props;
+
+    const nativeProps = getNativeProps<React.HTMLAttributes<HTMLElement>>(props, htmlElementProperties);
```

Comment:
Icons are a hot spot in perf of an app, as they can be rendered tons of times. I wonder if it makes sense to just avoid any unnecessaries altogether:

```tsx
// get all the things and have ...rest
const { color1, color2, color3, className, ...rest } = props;

return (
  <span 
    role="presentation"
    aria-hidden="true" 
    {...rest} 
    className={cx(classes, className)} 
  />);
```

Reply:
Yes but that would mean that we may end up with wrong props on the root, which is not the case with any other component in both libraries... Would rather merge it as is now and we can measure perf and decide in the next phase...